### PR TITLE
Change libdnet dependency from SF to github, change aufs-util from SF to github

### PR DESCRIPTION
**Pull request by paolomainardi**
Just some background information:

http://arstechnica.com/information-technology/2015/06/sourceforge-locked-in-projects-of-fleeing-users-cashed-in-on-malvertising/
https://plus.google.com/+gimp/posts/cxhB1PScFpe
http://helb.github.io/goodbye-sourceforge/
http://www.howtogeek.com/218764/warning-don%E2%80%99t-download-software-from-sourceforge-if-you-can-help-it/

I tried to recompile the iso using the fedoraproject.org package and everything seems ok.

### CODE REVIEW

**Comment by tianon** [comment]
This sounds fairly reasonable to me, but I'm a little bit concerned about the `04c394ed8e1e7fc455456e79e908916d` portion of that URL -- isn't this something that'll change with the next release of `libdnet`?  If so, do we have a reliable way to figure out what the "latest" version is?

Also, I'm a little bit concerned that I don't see any inkling of a pending move off sourceforge on either http://libdnet.sourceforge.net/ or http://sourceforge.net/projects/libdnet/ (nor any of their mailing lists).  Would perhaps a checksum to verify that the tarballs haven't been tampered with be a safer intermediate step before we abandon upstream's still-official release location outright?

**Comment by paolomainardi** [comment]
@tianon i've changed the repository from fedoraproject to googlecode and update version to 1.12, which it seems to be the only package available.

**Comment by SvenDowideit** [comment]
wow - and then there's https://github.com/boundary/libdnet

talk about old unmaintained SW deps

**Comment by simonvanderveldt** [comment]
FYI Arch uses the 1.12 release from Google Code as well https://www.archlinux.org/packages/community/x86_64/libdnet/

Note that there are commits from 2010 at the Google Code repository, but there hasn't been a release after 2007.

**Comment by paolomainardi** [comment]
@SvenDowideit we have to take in mind that google code will close 26th january 2016:
http://google-opensource.blogspot.com.tr/2015/03/farewell-to-google-code.html

Currently, i don't know which one is the "official" port of google code on github.

**Comment by simonvanderveldt** [comment]
Just scanned the mailing list @ sourceforge back till 2006, there's nothing there that says that the project is abandoned or moved to google code, only unanswered questions.

The same goes for Google Code, there's a issue there about Google Code closing with no response, just like all the other 30+ issues.
For all intents and purposes this project seems to be abandoned.

Do we really need it, open-vm-tools can be compiled [without it](https://github.com/vmware/open-vm-tools/blob/aaa20859c29247ca89a68f47c1a7b1a3b15bae09/open-vm-tools/configure.ac#L671).

I've reached out to the e-mail addresses I could find, will see if we can get an answer :)

**Comment by paolomainardi** [comment]
Thanks a lot Simon :)

Sorry for typos, sent by mobile.
Il 12/giu/2015 09:03 "Simon van der Veldt" <notifications@github.com> ha
scritto:

**Comment by tianon** [comment]
cc @frapposelli

**Comment by paolomainardi** [comment]
@tianon removed libdnet dependency, changed aufs-util from SF (which not responds anymore) and removed libdnet dependency.

**Comment by paolomainardi** [comment]
@tianon @SvenDowideit @simonvanderveldt

**Comment by frapposelli** [comment]
What's wrong with https://github.com/dugsong/libdnet ? that seems to be a mirror of what is on Google Code and maintained by the same guy, can't we just get that from there? am I missing something?

**Comment by paolomainardi** [comment]
@frapposelli you are right, let me change it again.

**Comment by paolomainardi** [comment]
@frapposelli updated.

**Comment by frapposelli** [comment]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -117,7 +117,7 @@ RUN curl -L http://http.debian.net/debian/pool/main/libc/libcap2/libcap2_2.22.or
 RUN cd /linux-kernel && \
     make INSTALL_HDR_PATH=/tmp/kheaders headers_install && \
     cd / && \
-    git clone http://git.code.sf.net/p/aufs/aufs-util && \
+    git clone https://github.com/Distrotech/aufs-util.git && \
```
Comment (simonvanderveldt) [inline Dockerfile]: According to the [announcement](https://webcache.googleusercontent.com/search?q=cache:3BwaFSfCUL4J:sourceforge.net/p/aufs/mailman/aufs-users/thread/18905.1431883360%40jrobl/+&cd=2&hl=nl&ct=clnk&gl=nl) all repo's moved to @sfjro except for the aufs-util repo :|

How certain are we that @distrotech is the right use/place to get the sources?
See [here](https://webcache.googleusercontent.com/search?q=cache:e_JMQzhpl4IJ:aufs.sf.net/+&cd=1&hl=nl&ct=clnk&gl=nl) for a cached version of the overview.

Also, Sourceforge is having an outage, it's not like they announced they would be shutting down like Google Code. So is it necessary to move away from Sourceforge?

File: Dockerfile [hunk 2]
```diff
@@ -197,7 +197,7 @@ RUN cd /vmtoolsd/open-vm-tools && \
     ./configure --disable-multimon --disable-docs --disable-tests --with-gnu-ld \
                 --without-kernel-modules --without-procps --without-gtk2 \
                 --without-gtkmm --without-pam --without-x --without-icu \
-                --without-xerces --without-xmlsecurity --without-ssl && \
+		--without-xerces --without-xmlsecurity --without-ssl && \
```
Comment (simonvanderveldt) [inline Dockerfile]: Guess this change is an accident?

**Comment by simonvanderveldt** [comment]
@frapposelli Do you know if libdnet is needed for the functionality that is used in boot2docker?
open-vm-tools can be compiled [without it](https://github.com/vmware/open-vm-tools/blob/aaa20859c29247ca89a68f47c1a7b1a3b15bae09/open-vm-tools/configure.ac#L671), which mean it can be dropped if not needed.

File: Dockerfile (hunk 1 above)
Comment (paolomainardi) [inline Dockerfile]: @simonvanderveldt yes i think that is necessary, read the links on my first post, they are rebasing code with spyware.

File: Dockerfile (hunk 2 above)
Comment (paolomainardi) [inline Dockerfile]: yes, let me fix it.

**Comment by SvenDowideit** [comment]
I'm going to test and merge this, as sf has been down for 3-4 days now.

**Comment by SvenDowideit** [comment]
mmm, ok, needs more testing, I'm getting a 

```
aufs au_opts_verify:1597:docker[793]: dirperm1 breaks the protection by the perm                        
                       ission bits on the lower branch 
```

when starting a container. idk if this is due to this change or not.

**Comment by SvenDowideit** [comment]
ok, it seems to work, we can modify later :)

**Comment by frapposelli** [comment]
@simonvanderveldt `libdnet` is needed for correct suspend/resume network actions in the guest so it shouldn't be ditched.

File: Dockerfile (hunk 1 above)
Comment (tianon) [inline Dockerfile]: I'm honestly very -1 on this change, especially since there's no verification or even hard-lock on a specific commit for this repo.

**Comment by tianon** [comment]
LGTM on the `libdnet` half, but the AUFS half needs more protections, especially an explicit commit checkout (which I'll go work up a PR to add now)

**Comment by paolomainardi** [comment]
@tianon i can open a new PR to hard lock a specific commit.

**Comment by tianon** [comment]
@paolomainardi that'd be perfect; I haven't started yet, so I'll let you take it from here :heart: :+1:

**Comment by paolomainardi** [comment]
@tianon check here https://github.com/boot2docker/boot2docker/pull/998

**Comment by tianon** [comment]
@paolomainardi thanks for the really quick turnaround! :heart: :+1:

**Pull request merged**